posts/2017-03-15-faunadb-serverless-authentication.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Using Serverless Authentication Boilerplate with FaunaDB
+description: FaunaDB is a security aware database, so we pass user-specific access tokens to it via a custom authorizer.
+date: 2017-03-15
+thumbnail: https://cloud.githubusercontent.com/assets/20538501/23813615/4b53e4fc-05a5-11e7-8214-e34c2c02b949.png
+layout: Post
+authors:
+ - ChrisAnderson
+---
+
+We are proud to support our friends at Fauna as they announce [FaunaDB Serverless Cloud](https://fauna.com/product), a globally consistent distributed database. This is a guest post from Chris Anderson, Director of Developer Experience, showing how to connect FaunaDB's access control model with Serverless platform best practices.
+
+> Forget the cloud — go Serverless with FaunaDB and let your pre-provisioned capacity woes become a thing of the past.
+> - **Chris Anderson, Fauna**
+
+A common serverless application architecture is to run an authentication service, which knows how to connect with OAuth identity providers like Facebook and Github, and exposes an authorizer lambda which can control access to your content functions. This makes authentication code reusable and cleanly separates it from your other functions.
+
+FaunaDB offers instance level security, so you can model your application’s data sharing patterns in the database. In this example we use the popular [`serverless-authentication-boilerplate`](https://github.com/laardee/serverless-authentication-boilerplate) to connect a FaunaDB app to Facebook login. Thanks to [Eetu Tuomala](https://www.linkedin.com/in/eetutuomala/) for the help with API Gateway details.
+
+In the spirit of the Serverless model, the authorizer supplies application content functions with a FaunaDB connection secret that corresponds to the currently logged-in user. This way, there is no possibility of application level bugs impacting data integrity and security.
+
+There is no limit to the data security patterns you can model in FaunaDB. See the tutorials for social graph examples, or follow this space for a multi-user TodoMVC example. For now, the content service just looks up the current user in the database.
```

Comment:
Should we add some links to the "social graph examples" and the "TodoMVC" example here?